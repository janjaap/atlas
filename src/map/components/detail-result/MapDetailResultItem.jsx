import React from 'react';
import PropTypes from 'prop-types';

const MapDetailResultItem = ({ label, value }) => value && (
  <li className="map-detail-result__item">
    <section className="map-detail-result__item-content">
      <div className="map-detail-result__item-label">{label}</div>
      <div className="map-detail-result__item-value">{value}</div>
    </section>
  </li>
);

MapDetailResultItem.defaultProps = {
  value: ''
};

MapDetailResultItem.propTypes = {
  label: PropTypes.string.isRequired,
  value: PropTypes.string
};

export default MapDetailResultItem;
