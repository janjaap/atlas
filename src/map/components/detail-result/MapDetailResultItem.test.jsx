import React from 'react';
import { shallow } from 'enzyme';

import MapDetailResultItem
  from './MapDetailResultItem';

describe('MapDetailResultItem', () => {
  it('should render label and value', () => {
    const label = 'label';
    const value = 'value';
    const wrapper = shallow(
      <MapDetailResultItem
        label={label}
        value={value}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should not render empty value', () => {
    const label = 'label';
    const value = '';
    const wrapper = shallow(
      <MapDetailResultItem
        label={label}
        value={value}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should not render without value', () => {
    const label = 'label';
    const wrapper = shallow(
      <MapDetailResultItem
        label={label}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
