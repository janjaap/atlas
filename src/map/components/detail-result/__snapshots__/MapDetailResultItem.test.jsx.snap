// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapDetailResultItem should not render empty value 1`] = `""`;

exports[`MapDetailResultItem should not render without value 1`] = `""`;

exports[`MapDetailResultItem should render label and value 1`] = `
<li
  className="map-detail-result__item"
>
  <section
    className="map-detail-result__item-content"
  >
    <div
      className="map-detail-result__item-label"
    >
      label
    </div>
    <div
      className="map-detail-result__item-value"
    >
      value
    </div>
  </section>
</li>
`;
